theurgy.lic

```diff
@@ -287,15 +287,21 @@ class Theurgy
     move 'go stair'
     waitfor 'Four beautifully detailed figures'
     walk_to '5846'
-    move 'pull candle'
-
-    case bput('study wall', 'Turning your attention to the sigils', 'interrupt your research')
-    when 'Turning your attention to the sigils'
-      waitfor 'as your understanding of the sigils gradually slips away.'
-    when 'interrupt your research'
-      echo 'Researching - skipping Study'
+    case bput('pull candle', 'You tug at the silver candlestick but it', 'You grasp hold of the silver candlestick and pull it back')
+    when 'You tug at the silver candlestick but it' 
+      walk_to '5756'
+      move 'go stair'
+      waitfor 'A low relief has been carefully carved'
+      walk_to '741'
+      return
+    when
+      case bput('study wall', 'Turning your attention to the sigils', 'interrupt your research')
+      when 'Turning your attention to the sigils'
+        waitfor 'as your understanding of the sigils gradually slips away.'
+      when 'interrupt your research'
+        echo 'Researching - skipping Study'
+      end
     end
-
     move 'go hatch'
```

Comment:
If you move this line inside the when on line 297, you can avoid duplicating all of the movement code on lines 292-296.